.github/workflows/pr-build.yml

```diff
@@ -7,7 +7,30 @@ on:
       - 'main'
 
 jobs:
-  run-pr-checks:
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+        pull-requests: read
+    outputs:
+      etl: ${{ steps.filter.outputs.etl }}
+      api: ${{ steps.filter.outputs.api }}
+      etl_tests: ${{ steps.filter.outputs.etl_tests }}
+      api_tests: ${{ steps.filter.outputs.api_tests }}
+    steps:
+    - uses: dorny/paths-filter@v3.0.2  
+      id: filter
+      with:
+        filters: |
+          etl:
+            - 'air-quality-backend/src/etl/**'
+          api:
+            - 'air-quality-backend/src/api/**'
+          etl_tests:
+            - 'air-quality-backend/tests/etl_tests/**'
+          api_tests:
+            - 'air-quality-backend/tests/api_tests/**'
```

Comment:
This is nice.

Should we combine etl + etl_tests together (and same for api). E.g.

```
          etl:
            - 'air-quality-backend/src/etl/**'
            - 'air-quality-backend/tests/etl_tests/**'
          api:
            - 'air-quality-backend/src/api/**'
            - 'air-quality-backend/tests/api_tests/**'
```

This way we don't need as big an if statement below - and I can't see any reason to keep them separate?